### PR TITLE
Feature/#5727/optional esc on close

### PR DESCRIPTION
**Pull request by Mickyfen17**
References #5727 
Add option to disable Escape key default close popup functionalty.

### CODE REVIEW

File: src/map/handler/Map.Keyboard.js [hunk 1]
```diff
@@ -162,7 +163,7 @@ export var Keyboard = Handler.extend({
 		} else if (key in this._zoomKeys) {
 			map.setZoom(map.getZoom() + (e.shiftKey ? 3 : 1) * this._zoomKeys[key]);
 
-		} else if (key === 27 && map._popup) {
+		} else if (key === 27 && mapPopup && mapPopup.options.closeOnEscapeKey) {
```
Comment (yohanboniface) [inline src/map/handler/Map.Keyboard.js]: Not sure we need an intermediate variable to save out 2 chars on two occurrences ;)

Comment (Mickyfen17) [inline src/map/handler/Map.Keyboard.js]: 😄 I went back and forth deciding whether or not to use that variable. I can remove it and just go back to `map._popup` if that's best?

Comment (yohanboniface) [inline src/map/handler/Map.Keyboard.js]: I'd say so, yes :)

Comment (Mickyfen17) [reply src/map/handler/Map.Keyboard.js]: Change made 🙂

File: src/map/handler/Map.Keyboard.js [hunk 2]
```diff
@@ -141,8 +141,8 @@ export var Keyboard = Handler.extend({
 		if (e.altKey || e.ctrlKey || e.metaKey) { return; }
 
 		var key = e.keyCode,
-		    map = this._map,
-		    offset;
+		map = this._map,
+		offset;
```
Comment (Motardo) [inline src/map/handler/Map.Keyboard.js]: Minor nitpick: Could you revert the white space here? It was 2 tabs of indentation plus 4 spaces of alignment. Thanks!

Comment (Mickyfen17) [reply src/map/handler/Map.Keyboard.js]: Spaces re-added.
For some reason when I first ran `jake test` I got 2 `Expected indentation of 2 tab characters but found 4` errors on line 144 & 145. Also linting errors in my text editor, therefore I just fixed the spacing as per the linter. 
Added back the 4 additional spaces of indentation and tests still passing.

**Comment by IvanSanchez** [comment]
Looks good. Thanks!

**Pull request merged**